tests/gold_tests/autest-site/cli_tools.test.ext

```diff
@@ -0,0 +1,47 @@
+'''
+Tools to help with TestRun commands
+'''
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+
+# example usage:
+# curlcmd = 'curl -vs -k --http2 https:127.0.0.1:{port}'.format(port = ts.Variables.port)
+#
+# tr = Test.AddTestRun()
+# tr.Processes.Default.Command = 'sleep 5'
+# tr.Processes.Default.ReturnCode = 0
+# tr.Processes.Default.StartAfter(
+#   *Test.ChainCurlCommands(*(numberOfRequests, curlcmd))
+#   )
+
+def make_curl_chain(tr, num, curlstr):
+    ret=[]
+
+    if "curl" not in curlstr:
+        return ret
+
+    for cnt in range(0,num):
+        ret.append(
+            tr.Processes.Process(
+                name="curl-{num}".format(num=cnt),
+                cmdstr=curlstr,
+                returncode = 0
+                )
+            )
+    return ret
+
+ExtendTest(make_curl_chain, name="ChainCurlCommands")
```

Comment:
* Why is this named `ChainCurlCommands`? We aren't chaining commands together, we are running one command in parallel a number of times independently.

* Do we need to document this extension so that other testers are aware it exists?

* It looks like we are starting this extension to help with various command line interface tools. This method seems to work only if a substring `curl` is found in the command line, and when it does not find it, it returns an empty list. This should be an error condition because otherwise it could result in a faulty test case.
   * In fact, if we want to make this a general utility, probably we should not specify `curl` at all, but run the given command it parallel, maybe as`ParallelCommand`?
   * Otherwise, if we intend to couple this to `curl`, should we require that `curl` available on the system? We won't necessarily be able to test for specific capabilities of `curl`—such as HTTP2—without parsing the command line more thoroughly.

Reply:
Good point. Restricting to `curl` doesn't really make sense. Changed to a generic `ParallelizeCommands` to allow for any command string to be duplicated and run. Yea, testing for the curl should be done with `Test.skipUnless...`.